libpromises/evalfunction.c

```diff
@@ -6021,7 +6021,11 @@ static FnCallResult FnCallReadData(ARG_UNUSED EvalContext *ctx,
     }
     else
     {
-        size_max = args->next ? IntFromString(RlistScalarValue(args->next)) : 0;
+        if (RlistLen(args) == 0)
+        {
+            FatalError(ctx, "Function '%s' requires at least one argument", fp->name);
```

Comment:
Of course you're right, we would not want cf-agent to die. I will close this one and submit a different PR, since I deleted my fork's branch thinking nobody likes the idea :) 

new PR: https://github.com/cfengine/core/pull/2417